### PR TITLE
Added FasterRCNN in its own subdirectory

### LINKED ISSUE #1310: Cleanup: move FasterRCNN to its own directory
more complex models probably need subdirectories - alternatively RetinaNet can be flattened in the OD directory.

### PR DESCRIPTION
**Pull request by jaygala223**
# What does this PR do?

Fixes #1310 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link to it if that's the case. 
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
Hello @LukeWood, can you please review this PR?

### CODE REVIEW

**Comment by LukeWood** [comment]
Thanks for the PR @jaygala223!

A few comments:

1.) you have a merge conflict
2.) you'll need to update keras_cv/models/__init__.py to reference the new location
3.) you'll need to make sure all tests pass.

Thanks for your help!

**Comment by jaygala223** [comment]
Thank you for the feedback, Luke. Working on it 👍

**Comment by jaygala223** [comment]
Hi @LukeWood, the tests are passing now and I updated models/__init__.py to reference the new location. Can you help me resolve the merge conflicts and the failing checks? I am not quite able to understand what's wrong 🤔

image

File: keras_cv/__init__.py [hunk 1]
```diff
@@ -23,9 +23,10 @@
 from keras_cv import layers
 from keras_cv import losses
 from keras_cv import metrics
-from keras_cv import models
+# from keras_cv import models
```
Comment (ianstenbit) [inline keras_cv/__init__.py:26]: Looks like this accidentally got commented out?

File: keras_cv/__init__.py [hunk 2]
```diff
@@ -23,9 +23,10 @@
...
 from keras_cv import training
 from keras_cv import utils
+from keras_cv.models import FasterRCNN
```
Comment (ianstenbit) [inline keras_cv/__init__.py:29]: Let's not directly import FasterRCNN from the top-level init

File: keras_cv/models/object_detection/faster_rcnn/__init__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2022 The KerasCV Authors
```
Comment (ianstenbit) [inline keras_cv/models/object_detection/faster_rcnn/__init__.py:1]: nit: 2023

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks for the PR!

Looks good to me except for a few little changes.

@LukeWood can you verify that we want to move forward with this approach?

File: keras_cv/__init__.py (hunk 1 above)
Comment (jaygala223) [reply keras_cv/__init__.py:26]: Oops! Yes, this got commented out by accident. Fixing it

File: keras_cv/__init__.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/__init__.py:29]: Users can import it as `keras_cv.models.FasterRCNN`

With this line included, users would import it as `keras_cv.FasterRCNN`, which we don't want to support.

Comment (jaygala223) [reply keras_cv/__init__.py:29]: Oh right. I get it now.

Comment (ianstenbit) [reply keras_cv/__init__.py:29]: yep! :)

Comment (jaygala223) [reply keras_cv/__init__.py]: Great. Thanks!

**Review by LukeWood — APPROVED** [review]

**Comment by jaygala223** [comment]
Yes @LukeWood, I am working on it

**Comment by jaygala223** [comment]
Hi @LukeWood, I'm unable to see these conflicts on my system. Can you please help me out?

**Comment by jaygala223** [comment]
Sorry, I accidentally discarded all the commits. Fixing it now

**Comment by jaygala223** [comment]
I think I'll make a new branch and submit the changes because it's too complicated here. Sorry for the trouble.

**Pull request closed without merging**